### PR TITLE
Feature/joinpods config

### PR DESCRIPTION
**Pull request by christophe-duc**
A simple set_pods_config function that can allow the user to configure following or commenting inside the pod

### CODE REVIEW

File: instapy/instapy.py [hunk 1]
```diff
@@ -5272,6 +5274,20 @@ def accept_follow_requests(self,
 
         return self
 
+    def set_pods_config(self,skip_comments: bool=False, skip_follows: bool=False):
+        """ Function to configure pods"""
+        if skip_comments is True:
+            self.skip_pod_comments = True
+        if skip_comments is False:
```
Comment (breuerfelix) [inline instapy/instapy.py]: what about self.skip_pod_comments = skip_comments ?

File: instapy/instapy.py [hunk 2]
```diff
@@ -5272,6 +5274,20 @@ def accept_follow_requests(self,
...
+            self.skip_pod_comments = False
+
+        if skip_follows is True:
+            self.skip_pod_follows = True
```
Comment (breuerfelix) [inline instapy/instapy.py]: what about self.skip_pod_follows = skip_follows ?
or if you also wanna have the None convered:
`self.skip_pod_follows = skip_follows or False`

File: instapy/instapy.py (hunk 1 above)
Comment (christophe-duc) [reply instapy/instapy.py]: yes, much simpler and with the type_checks it should take care of case for None

File: instapy/instapy.py (hunk 2 above)
Comment (christophe-duc) [reply instapy/instapy.py]: fixed in new commit

**Comment by breuerfelix** [comment]
I just added or False no the value can never be None :)

**Review by breuerfelix — DISMISSED** [review]
I like it!

**Comment by ishandutta2007** [comment]
This feature should not be allowed at all.
Maybe you have not understood the idea of pods in the first place.
I thought you were talking about skipping comments for your own post.
But here your are skipping comments and following for the posts you are downloading. In other words you want to get your work done by uploading yours but not help others by engaging with them.
If everyone runs with such skip setting, pod posts will receive zero engagements.

You are welcome to write features configuring engagement for your posts, but not skip following or commenting for others post if the other person has not opted out.

**Comment by christophe-duc** [comment]
> This feature should not be allowed at all.
> Maybe you have not understood the idea of pods in the first place.
> I thought you were talking about skipping comments for your own post.
> But here your are skipping comments and following for the posts you are downloading. In other words you want to get your work done by uploading yours but not help others by engaging with them.
> If everyone runs with such skip setting, pod posts will receive zero engagements.
> 
> You are welcome to write features configuring engagement for your posts, but not skip following or commenting for others post if the other person has not opted out.

I think my understanding is a little bit different, but let's try to discuss this in a civil way.

1. Activating commenting or not: if you look at posts participating in the pod, you will see that they have 10's of comments that are the same, as the average instapy users always use the basic comments. In terms of engagement I think this is quite bad as it shows right away you are participating in an automated pod

2. Activating following or not: again as this an automated pod, we follow and then unfollow. For example I am not very interested by being followed by a bot, which will unfollow me a couple days laters. I agree that here my idea should go a little bit farther in the sense, that I should also be able to say in the pod that I am not interested to be followed (so the other participants in the pod don't waste time on that)

3. Liking: for me that the real interest of a pod, is to get my post into the top posts of a certain hash tag, because it is liked a lot. I think your current vision of the pod, does not allow this, and this is why I am not using it, and am trying to propose solution, so it's more usable to the rest of us.

Cheers.

**Comment by breuerfelix** [comment]
In my opinion, this config should be done at the join pods method.

If you join a pod you can specify liking, following or commenting (or more than one).

Then you just follow people for example and only get follows back. So if you decide that commenting and following is nothing special, you can just join pods for the liking feature.

**Comment by ishandutta2007** [comment]
@christophe-duc Point 1,2 3, whatever your interests is just configure to upload your post with all those three configuration changes. you have no right to override other peoples configurations. I don't think I can explain it any better.

**Comment by breuerfelix** [comment]
@ishandutta2007 but you should be able to decide what people are going to write to YOUR posts.

So you should be able to disable commenting / following or liking at all.

Right now its only all or nothing. Thats okay for the start but I can understand people don't want stupid comments so they don't it at all.

**Comment by ishandutta2007** [comment]
ok see if this helps: 
current upload format is 

`{post_id, 
commenting_mode(10% or 30% or 90%)}
`

so to make it fully configurable it should be 
`{post_id, 
commenting_mode(0% or 10% or 30% or 90%), 
liking_mode(0% or default%), 
follow_me(True(default) or False)}`

**Comment by ishandutta2007** [comment]
> Right now its only all or nothing. Thats okay for the start but I can understand people don't want stupid comments so they don't it at all.

Yes that why the 4th mode to opt out from receiving comments.

**Comment by ishandutta2007** [comment]
> but you should be able to decide what people are going to write to YOUR posts.

there was a thread on creating a global templates of 400-500 comments for each of the topics on CDN which people will download and use while interacting with posts of that topic so that people don't receive any user created junk.

**Comment by christophe-duc** [comment]
> ok see if this helps you understand:
> current upload format is
> 
> `{post_id, commenting_mode(10% or 30% or 90%)} `
> 
> so to make it fully configurable it should be
> `{post_id, commenting_mode(0% or 10% or 30% or 90%), liking_mode(0% or default%), follow_me(True(default) or False)}`

So this is a step in the right direction, that works.

Still doesn't solve the fact that I have to surrender the actions of my bot, create comments (which I simply never use on my bot) to participate in the pod. If I don't want my bot to follow or comments on others, that should be my decision.

So I still think people should have the rights to control what their bot do (commenting and following) and also have what @ishandutta2007 propose: `{post_id, commenting_mode(0% or 10% or 30% or 90%), liking_mode(0% or default%), follow_me(True(default) or False)}`

with the 2 sides (being: to configure what your bot do and what you receive), it becomes quite good and as always the community will find a balance.

This PR, just provides the "configure what your bot do".

**Comment by ishandutta2007** [comment]
> Still doesn't solve the fact that I have to surrender the actions of my bot, create comments (which I simply never use on my bot) to participate in the pod. If I don't want my bot to follow or comments on others, that should be my decision.

That's the exact point of pods, I guessed it right at the beginning itself that you haven't read about pods before jumping onboard, in Pods **IT IS NOT YOUR DECISION IF YOU WANT TO ENGAGE OR  NOT**. In properly moderated pods until you have engaged enough(eg say 500 likes given and 50 comments made) with your fellow pod members posts, you are not even allowed to share your own post. People who try to act like you mentioned are termed "LEECHERS" and are thrown out of pods. So when you read about PODs do read about **_leechers_** too

**Comment by christophe-duc** [comment]
@ishandutta2007 : again I don't really understand why you feel the need to yell at me, I am again having a civilized discussion.

- the bot do participate in the pod as it will be providing likes, it will give you 3,000 likes instead of 500 comments if it doesn't have to comment. How is that going to be leeching, if it does help you being in the top post of a tag?

if there is some kind of value system you want to enforce, feel free, 1000 likes = 1 comments, no problem. In any case, if you really want to enforce things (and that's also a good idea), then an external process should scan the participating bots database and verify that they really do what they say they would do (as far as I see, in the code, there is no prevention for somebody external to the system to upload a post in the firebase dabatase and have all the bots working on it without himself participating actively in it. Everybody has the source code and can patch their local version of instapy to do what they want to do.

If you want a good system let's design it well so abuses are not possible.

Closing this, I don't like the yelling part.

**Comment by ishandutta2007** [comment]
> yell at me

Ahh caps here was not yelling. it was just to highlight the central point of discussion.. relax.
and have a good day

**Comment by converge** [comment]
It's open source, let's share the love S2.

Pods are a great feature, but I also don't agree about do comments on pods, but let's keep the conversation open and respectful for new contributors, our only goal is a better tool for everyone.

I'll have a closer look tomorrow and try to help the discussion somehow.

**Comment by ishandutta2007** [comment]
> agree about do comments

what about comments tell me your views ?

**Comment by ishandutta2007** [comment]
@converge let's know when you are available for the discussion, just my point other day was in no way to disrespect anyone but to let new people read about past conversation, issues and background before a discussion. I am anytime available for discussion. In fact @christophe-duc made a valid point to build a hack-proof process via eternal scanning etc, but he is unaware that we had discussed all those before and these action items to build scanning, tracking seed metrics and authenticating etc are all discussed and tracked under separate issues.

**Comment by ishandutta2007** [comment]
> In my opinion, this config should be done at the join pods method.

@breuerfelix  It is at joining of pods only 
https://github.com/timgrossmann/InstaPy/blob/d4d57a70c60f1b2ca08d6d2c480f1e4cd2aa2ae0/instapy/instapy.py#L5275

engagement_mode='normal' means 30% of active pod members with comment on my posts.
engagement_mode='light' means 10% of active pod members with comment on my posts.

**Review by MedRedha — APPROVED** [review]

**Comment by ishandutta2007** [comment]
@MedRedha can you explain this ?

**Review by ishandutta2007 — CHANGES_REQUESTED** [review]
Please do these [three](https://github.com/timgrossmann/InstaPy/pull/4679#issuecomment-512881023) changes for posts being uploaded and not posts downloaded as it violates the basic principles of pods and resubmit the PR

**Comment by converge** [comment]
finally here, sorry for the delay.

Can we just disable commenting and move on?

**Comment by ishandutta2007** [comment]
> finally here, sorry for the delay.
> 
> Can we just disable commenting and move on?

That's what I have been saying from line one of this entire thread. I don't know why it is so difficult to understand. I tried so hard to make that guy understand this simple thing, Just you have to add a 4th level and resubmit. Just one line change. If you just want to disable commenting, you have set the param  and call `session.join_pods(mode="no_comments")` and you will receive zero comments.

**Comment by converge** [comment]
what if I don't want to receive comments from Pods?

**Comment by ishandutta2007** [comment]
this means you wont receive any comments at all ,not a single one in entire lifetime of using instapy. Please read my messages properly , zero means zero.

**Comment by ishandutta2007** [comment]
> and you will receive zero comments.

I already mentioned here .

**Comment by converge** [comment]
the point is, most users don't want to receive random comments, that's why we should disable it, don't you think?

**Comment by ishandutta2007** [comment]
Yes but the few who wants,  let them have it.

**Comment by ishandutta2007** [comment]
I dont know if my english is different from yours. This is the first time I am seeing this kind of miscommunication. I will make the change and submit . wait 5 mins and review the change

**Comment by ishandutta2007** [comment]
@converge plese CR
https://github.com/timgrossmann/InstaPy/pull/4865
https://github.com/InstaPy/instapy-quickstart/pull/70

**Comment by converge** [comment]
> I dont know if my english is different from yours.

That's my breakpoint, I will always prefer to use my time to share the love, and I won't continue the talk with this kind of insult. peace 💌

**Comment by daliborg** [comment]
I vote for disabling comments.

**Comment by breuerfelix** [comment]
I will close this PR since @ishandutta2007 opened a new one which disabled commenting :)

**Pull request closed without merging**